widget/view.controller.js

```diff
@@ -4,76 +4,104 @@
-        heatmapService.executeAction(_connectorName, _connectorAction, payload).then(function (response) {
-          if(response && response.data && response.data.length > 0)
+        $q.all(_promises).then(function(response){
+          $scope.processing = false;
+          if(response[0] && response[0].data && response[0].data.length > 0)
           {
-            setDistributionData(response.data);
+            setDistributionData(response[0].data);
           }
-        });
+           if(response[1] && response[1].data)
+          {
+            $scope.connectorOperationResponse = response[1].data;
+            setMultipleFieldsData($scope.connectorOperationResponse);
+          } 
+        })
       }
+      else{
+        executeAPIQuery();
+      }  
+    }
 
-      function executeAPIQuery(){
-        heatmapService.getResourceAggregate($scope.config).then(function (response) {
-          if (response && response.data['hydra:member'] && response.data['hydra:member'].length > 0) {
-            setDistributionData(response.data['hydra:member']);
-          }
-        }, function (error) {
-          console.log(error);
-        });
+    function checkCurrentPage(state){
+      if (state.current.name.includes('viewPanel.modulesDetail')) {
+        let params = $scope.pageState.current.params;
+        $scope.indicator = params.id;
       }
+    }
 
-      function setDistributionData(_responseData){
-        $scope.distribution = angular.copy(_responseData);
-        $scope.distributionSegment = 100 / _responseData.length; //width of each section
-        $scope.processing = false;
-      }
+    function executeConnectorAction(_moduleMetaData){ 
+      let _connectorName = _moduleMetaData.connector;
+      let _connectorAction = _moduleMetaData.actions.risk_distribution;
+      let payload = { 'indicator': $scope.indicator || '8.8.8.8' };
+      return heatmapService.executeAction(_connectorName, _connectorAction, payload)
+    }
+
+    function executeConnectorOperation(_moduleMetaData){
+      let _connectorName = _moduleMetaData.connector;
+      let _connectorAction = _moduleMetaData.operation;
+      let payload = { 'indicator': $scope.indicator || '8.8.8.8' };
+      return heatmapService.executeAction(_connectorName, _connectorAction, payload);
+    }
+
+    function executeAPIQuery(){
+      heatmapService.getResourceAggregate($scope.config).then(function (response) {
+        if (response && response.data['hydra:member'] && response.data['hydra:member'].length > 0) {
+          setDistributionData(response.data['hydra:member']);
+        }
+      }, function (error) {
+        console.log(error);
+      });
+    }
+
+    function setDistributionData(_responseData){
+      $scope.distribution = angular.copy(_responseData);
+      $scope.distributionSegment = 100 / _responseData.length; //width of each section
+      $scope.processing = false;
+    }
 
-      init();
+    function setMultipleFieldsData(_operationData){
```

Comment:
can we call this a titleFieldsData? or summaryFieldsData?

Reply:
update to titleFieldsData